pkg/router/base_rule.go

```diff
@@ -180,15 +183,37 @@ func (rri *RouteRuleImplBase) matchRoute(headers api.HeaderMap, randomValue uint
 }
 
 func (rri *RouteRuleImplBase) finalizePathHeader(headers api.HeaderMap, matchedPath string) {
-	if len(rri.prefixRewrite) < 1 {
+
+	if len(rri.prefixRewrite) < 1 && len(rri.regexRewrite.Pattern.Regex) < 1 {
 		return
 	}
+
 	if path, ok := headers.Get(protocol.MosnHeaderPathKey); ok {
-		if strings.HasPrefix(path, matchedPath) {
-			headers.Set(protocol.MosnOriginalHeaderPathKey, path)
-			headers.Set(protocol.MosnHeaderPathKey, rri.prefixRewrite+path[len(matchedPath):])
-			log.DefaultLogger.Infof(RouterLogFormat, "routerule", "finalizePathHeader", "add prefix to path, prefix is "+rri.prefixRewrite)
+
+		//If both prefix_rewrite and regex_rewrite are configured
+		//prefix rewrite by default
+		if len(rri.prefixRewrite) > 1 {
+			if strings.HasPrefix(path, matchedPath) {
+				headers.Set(protocol.MosnOriginalHeaderPathKey, path)
+				headers.Set(protocol.MosnHeaderPathKey, rri.prefixRewrite+path[len(matchedPath):])
+				log.DefaultLogger.Infof(RouterLogFormat, "routerule", "finalizePathHeader", "add prefix to path, prefix is "+rri.prefixRewrite)
+			}
+			return
+		}
+
+		// regex rewrite path if configured
+		if len(rri.regexRewrite.Pattern.Regex) > 1 {
+			reg := regexp.MustCompile(rri.regexRewrite.Pattern.Regex)
```

Comment:
if `rri.regexRewrite.Pattern.Regex`  cannot be parsed will caused `MustCompile` panic ?

Reply:
i am sorry about that, maybe we should check regex in mosn just started. if got the error, panic mosn? 

like:  

```
reg,err  := regexp.Compile("\\  ? as")
if err!=nil{
   panic(err)
}
```

Reply:
Could  printe a error-level log  to alert the user.